hydra-node/src/Hydra/Options.hs

```diff
@@ -305,22 +305,33 @@ chainConfigParser =
 
 networkIdParser :: Parser NetworkId
 networkIdParser =
-  testnetParser
+  option
+    (parseTestnet <|> parseMainnet)
+    ( long "network-id"
+        <> showDefault
+        <> value (Testnet (NetworkMagic 42))
```

Comment:
The shown default will not be an accepted value to the parser. Usually you would use `showDefaultWith` to modify that.